hera_cal/tests/test_io.py

```diff
@@ -226,12 +226,39 @@ def test_load_cal(self):
         self.assertEqual(times.shape, (3,))
         self.assertEqual(sorted(pols), ['x','y'])
 
-
-    #TODO: implement this test
     def test_write_cal(self):
-        with self.assertRaises(NotImplementedError):
-            io.write_cal()
-
+        # create fake data
+        ants = np.arange(10)
+        pols = np.array(['x'])
+        freqs = np.linspace(100e6, 200e6, 64, endpoint=False)
+        Nfreqs = len(freqs)
+        times = np.linspace(2458043.1, 2458043.6, 100)
+        Ntimes = len(times)
+        gains = {}
+        quality = {}
+        flags = {}
+        for i, p in enumerate(pols):
+            for j, a in enumerate(ants):
+                gains[(a, p)] = np.ones((Ntimes, Nfreqs), np.complex)
+                quality[(a, p)] = np.ones((Ntimes, Nfreqs), np.float) * 2
+                flags[(a, p)] = np.zeros((Ntimes, Nfreqs), np.bool)
+
+        # test basic execution
+        uvc = io.write_cal("ex.calfits", gains, freqs, times, pols, flags=flags, quality=quality,
+                           overwrite=True, return_uvc=True, write_file=True)
+        self.assertTrue(os.path.exists("ex.calfits"))
+        self.assertAlmostEqual(uvc.gain_array[0,0,0,0,0], (1+0j))
```

Comment:
Do you want to loop over all antennas and pols and do these assertions?
I understand it's redundant, but there may be some weird edge cases not found by this test.

Reply:
meh, not a fan of the idea of looping through the whole array... instead of looping over I just checked the sum of the entire array is what I would expect